AppDevPods/AppDevCameraKit/ADKCamera.m

```diff
@@ -0,0 +1,1342 @@
+            videoCaptureConnection.preferredVideoStabilizationMode = stabilizationMode;
+        }
+    }
+
+    self.captureStillImageOutput.lensStabilizationDuringBracketedCaptureEnabled = stabilization;
+    AVCaptureConnection *photoConnection = [self.captureStillImageOutput connectionWithMediaType:AVMediaTypeVideo];
+    if (photoConnection.isVideoStabilizationSupported) {
+        photoConnection.preferredVideoStabilizationMode = stabilizationMode;
+    }
+}
+
+
+#pragma mark -- Private supported methods
+
+- (AVCaptureDevice *)captureDeviceWithCameraPosition:(AVCaptureDevicePosition)cameraPosition
+{
+    NSArray *devices = [AVCaptureDevice devicesWithMediaType:AVMediaTypeVideo];
+    for (AVCaptureDevice *device in devices) {
+        if (device.position == cameraPosition) {
+            return device;
+        }
+    }
+
+    return nil;
+}
+
+- (AVCaptureConnection *)captureConnectionWithCaptureOutput:(AVCaptureOutput *)expactedCaptureOutput
+{
+    AVCaptureConnection *captureConnection = nil;
+    for (AVCaptureConnection *connection in expactedCaptureOutput.connections) {
+        for (AVCaptureInputPort *inputPort in connection.inputPorts) {
+            if ([inputPort.mediaType isEqual:AVMediaTypeVideo]) {
+                captureConnection = connection;
+                break;
+            }
+        }
+    }
+
+    return captureConnection;
+}
+
+- (AVCaptureVideoOrientation)captureVideoOrientation
+{
+    AVCaptureVideoOrientation orientation;
+
+    if(self.alignDeviceOrientation) {
+        switch ([UIDevice currentDevice].orientation) {
+            case UIDeviceOrientationLandscapeLeft:
+                orientation = AVCaptureVideoOrientationLandscapeRight;
+                break;
+            case UIDeviceOrientationLandscapeRight:
+                orientation = AVCaptureVideoOrientationLandscapeLeft;
+                break;
+            case UIDeviceOrientationPortraitUpsideDown:
+                orientation = AVCaptureVideoOrientationPortraitUpsideDown;
+                break;
+            default:
+                orientation = AVCaptureVideoOrientationPortrait;
+                break;
+        }
+    } else {
+        switch ([UIApplication sharedApplication].statusBarOrientation) {
+            case UIInterfaceOrientationLandscapeLeft:
+                orientation = AVCaptureVideoOrientationLandscapeLeft;
+                break;
+            case UIInterfaceOrientationLandscapeRight:
+                orientation = AVCaptureVideoOrientationLandscapeRight;
+                break;
+            case UIInterfaceOrientationPortraitUpsideDown:
+                orientation = AVCaptureVideoOrientationPortraitUpsideDown;
+                break;
+            default:
+                orientation = AVCaptureVideoOrientationPortrait;
+                break;
+        }
+    }
+
+    return orientation;
+}
+
+- (void)updateCameraStatus
+{
+    self.cameraMirrorMode = _cameraMirrorMode;
+    self.cameraFlashMode = _cameraFlashMode;
```

Comment:
If you change the lens, you need to update camera settings for satisfying functional requirements.